### ISSUE TITLE
QuickEdit UI for box-shadow

### ISSUE THREAD

**Issue by flukeout** [mozilla/thimble.mozilla.org#2508]
It would be great to have a QuickEdit UI (similar to how we handle the colorpicker) for the CSS property ``box-shadow``. The goal would be to create an expandable UI that would allow people to set the shadow property, which is slightly complicated...

Here's a very rough mockup of the UI controls it could have...

```
Left offset:     -----O------ 5px
Top offset:    ---------O-- 7px
Blur Radius: O----------- 8px

Color:  [ Basic Colorpicker ]
```

Since the ``box-shadow`` property includes a color we could do a couple of things:

* See if we could re-use / integrate the existing colorpicker
  * We could maybe reduce some of its complexity?
* Include just the color black as an ``rgba(0,0,0,1)`` value
  * Allow the user to control the alpha value with a slider

If we can get these basic controls working, then we can move to a secondary phase of improving the look and feel of the UI.

**Comment by Cryzek** [comment]
Will this require creating an extension in brackets, like the one available for color editor, or is there a different way to do this?
The brackets docs mention that we "cannot load modules from different extensions."

Also, currently while editing `color` value in `box-shadow` property, color editor appears automatically. See below
![capture1](https://user-images.githubusercontent.com/9551285/31172152-136e31a4-a920-11e7-803d-dad3346392c9.PNG)
![capture2](https://user-images.githubusercontent.com/9551285/31172154-14a607d6-a920-11e7-9706-b5f8d75180ca.PNG)

**Comment by humphd** [comment]
@Cryzek I wrote some thoughts on this in a similar bug, see https://github.com/mozilla/thimble.mozilla.org/issues/2506#issuecomment-333935686

**Comment by flukeout** [comment]
@Cryzek - yeah, we'd have to figure out how to handle cases like the one you found. I guess if we can detect that you are within the value for a ``box-shadow`` property, then we can show the Box Shadow UI instead of just the color UI.

**Comment by Cryzek** [comment]
Thanks @humphd, @flukeout . Can I try this one? Would be awesome to build an extension.

**Comment by Cryzek** [comment]
Hi. Can this one be assigned to me? I'll require a little more time than I thought to get this working.

**Comment by flukeout** [comment]
Yeah, awesome @Cryzek - sorry for being slow to chime in on the PR. We are getting ready for the mozilla festival, so I've been busy helping with that. Let us know if you have any questions, etc.

**Comment by gideonthomas** [comment]
@Cryzek are you still working on / interested in this issue?

**Comment by Cryzek** [comment]
Yes, I'm interested in this and will see to it's completion. I looked up on the InlinePaddingEditor work, and tried to use a similar approach , but it didn't work. So, I'm trying a different method. I'm sorry for the delays.

**Comment by gideonthomas** [comment]
No worries at all. Let us know if you need any help. The quick edit for margin or padding: https://github.com/mozilla/brackets/pull/918 should be similar to what you're going to implement, except it won't have corners nor will it have an all corners option.
